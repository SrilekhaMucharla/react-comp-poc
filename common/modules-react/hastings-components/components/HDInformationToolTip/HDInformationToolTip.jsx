/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { useEffect } from 'react';
// import './hd-information-tooltip.scss';
import { PropTypes } from 'prop-types';

function HDInformationToolTip(props) {
    const {
        heading, content, hidden
    } = props;
    const [modalShow, setModalShow] = React.useState(false);
    useEffect(() => {
        if (modalShow) {
            document.body.classList.add('hd-modal-open');
        } else {
            document.body.classList.remove('hd-modal-open');
        }
    }, [modalShow]);

    return (
        <div className="hd-information-tooltip">
            <button hidden={hidden} type="button" className="hd-tooltip" onClick={() => setModalShow(true)}>
                <i className="fa fa-info-circle" aria-hidden="true" />
            </button>
            {modalShow
                ? (
                    <div className="modal">
                        <div className="modal-outside" onClick={() => setModalShow(false)} />
                        <div className="modal-container">
                            <div className="modal-header">
                                {heading}
                                <div className="modal-close">
                                    <div onClick={() => setModalShow(false)}><i className="fa fa-close" aria-hidden="true" /></div>
                                </div>
                            </div>
                            {/* <hr /> */}
                            <div className="modal-body">{content}</div>
                        </div>
                    </div>
                ) : ''}
        </div>
    );
}

HDInformationToolTip.propTypes = {
    heading: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    hidden: PropTypes.bool
};

HDInformationToolTip.defaultProps = {
    hidden: false
};

export default HDInformationToolTip;
