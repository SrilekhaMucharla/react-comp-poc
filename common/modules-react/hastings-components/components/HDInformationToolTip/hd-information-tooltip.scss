.hd-information-tooltip  { 
    .modal {
        position: fixed;
        left: 0;
        top: 0;
        width: 100%;
        height: 100%;
        background: rgba(0,0,0,0.5);
        display: inline-flex;
        align-items: flex-end;
        justify-content: flex-end;

        .modal-container{
            background-color: $white-fill;
            border-radius: 5px;
            height:100%;
            width: 40%;
            @include media-breakpoint-between(xs, sm) {
                width: 100%;
                border-radius: 0;
            }
        }

        .modal-outside {
            background: rgba(0,0,0,0.5);
            position: fixed;
            left: 0;
            top: 0;
            width: 100%;
            height: 100%;
            z-index: -1;
        }

        .modal-body {
            height:100%;   
            padding-left: 5px;         
        }

        .modal-header {
            padding-left: 5px;
            padding-top: 5px;
        }

        .modal-close {
            float: right;
            margin-right:10px;            
        }     
    }
}

//TODO: find the right for below
.mobile {
    .modal-container {        
        width: 70%;
    }
}

.hd-modal-open {
    overflow: hidden;
}

.hd-tooltip {
    background: none!important;
    border: none;
    padding: 0!important;        
    text-decoration: underline;
    cursor: pointer;
}
