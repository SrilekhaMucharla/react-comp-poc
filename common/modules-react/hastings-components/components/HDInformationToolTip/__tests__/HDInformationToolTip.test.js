import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDInformationToolTip from '../HDInformationToolTip';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDInformationToolTip />', () => {
    it('render information tooltip', () => {
        const wrapper = shallow(<HDInformationToolTip id="abcd" heading="Hello heading" content="Hello Content" />);
        expect(wrapper).toMatchSnapshot();
    });
});
