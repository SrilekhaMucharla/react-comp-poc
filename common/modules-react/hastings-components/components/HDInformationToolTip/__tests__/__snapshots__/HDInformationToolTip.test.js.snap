// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDInformationToolTip /> render information tooltip 1`] = `
<div
  className="hd-information-tooltip"
>
  <button
    className="hd-tooltip"
    hidden={false}
    onClick={[Function]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-info-circle"
    />
  </button>
</div>
`;
